src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java

```diff
@@ -220,14 +246,14 @@ private void initButtons() {
             int dx = (i % 2) * 8;
             int dy = (i / 2) * 8;
 
-            ColorSwatchWidget widget = addDrawableChild(new ColorSwatchWidget(
+            ColorSwatchWidget swatch = new ColorSwatchWidget(
                     Text.translatable("text.scribble.color." + color.getName()), color,
-                    () -> {
-                        this.getRichSelectionManager().setColor(color);
-                        this.setSwatchColor(color);
-                    }, x + 3 + dx, y + 95 + dy, 8, 8
-            ));
+                    () -> changeActiveColor(color),
+                    x + 3 + dx, y + 95 + dy, 8, 8
+            );
+            swatch.setToggled(activeColor == color);
```

Comment:
Nitpick, feel free to ignore: if we're always setting this anyways, we can also just add this as a constructor parameter.